Fix #6898: Replace atoi() with strtoul()

MiguelHorta opened the pull request:
Normalize type and parsing of generation_seed across all files
Add static_assert() to ensure correct type

LordAro reviewed (dismissed):
Looks good - you might want to check your git config, github seems to think the commit has been authored/committed by 2 different people, neither of which is linked to your GH account

MiguelHorta committed:
Fix #6898: Replace atoi() with strtoul()

Normalize type and parsing of generation_seed across all files
Add assert_compile() to ensure correct type

MiguelHorta commented:
Thanks for the head-ups. Indeed, my git configuration was wrong.
Unfortunately, it seems I invalidated your review while trying to fix, sorry about that :|
